device/src/web.ts

```diff
@@ -106,6 +113,54 @@ export class DeviceWeb extends WebPlugin implements DevicePlugin {
       uaFields.operatingSystem = 'unknown';
     }
 
+    // Check for browsers based on non-standard javascript apis, only not user agent
+    const isFirefox = !!window.InstallTrigger;
+    const isSafari = !!window.ApplePaySession;
+    const isChrome = !!window.chrome;
+    const isEdge = /Edg/.test(ua);
+    const isFirefoxIOS = /FxiOS/.test(ua);
+    const isChromeIOS = /CriOS/.test(ua);
+    const isEdgeIOS = /EdgiOS/.test(ua);
+
+    // FF and Edge User Agents both end with "/MAJOR.MINOR"
+    if (
+      isSafari ||
+      (isChrome && !isEdge) ||
+      isFirefoxIOS ||
+      isChromeIOS ||
+      isEdgeIOS
+    ) {
+      // Safari version comes as     "... Version/MAJOR.MINOR ..."
+      // Chrome version comes as     "... Chrome/MAJOR.MINOR ..."
+      // FirefoxIOS version comes as "... FxiOS/MAJOR.MINOR ..."
+      // ChromeIOS version comes as  "... CriOS/MAJOR.MINOR ..."
+      let searchWord: string;
+      if (isFirefoxIOS) {
+        searchWord = 'FxiOS';
+      } else if (isChromeIOS) {
+        searchWord = 'CriOS';
+      } else if (isEdgeIOS) {
+        searchWord = 'EdgiOS';
+      } else if (isSafari) {
+        searchWord = 'Version';
+      } else if (isChrome) {
+        searchWord = 'Chrome';
+      }
+
+      const words = ua.split(' ');
+      words.forEach(word => {
```

Comment:
`.forEach` is discouraged. (It's a remnant of pre-ES6 days when `for..of` wasn't a thing).

`for (const word of words) {` is preferred 😊